hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java

```diff
@@ -127,18 +133,24 @@ public void doIn(@NonNull final TransferContext transferContext) {
      * @param accountStore      The account store
      * @param tokenRelStore     The token relation store
      * @param handleContext     The context
+     * @param amount            The amount of tokens to be transferred
      */
     private TokenAssociation validateAndBuildAutoAssociation(
             @NonNull final AccountID accountId,
             @NonNull final TokenID tokenId,
             @NonNull final Token token,
             @NonNull final WritableAccountStore accountStore,
             @NonNull final WritableTokenRelationStore tokenRelStore,
-            @NonNull final HandleContext handleContext) {
+            @NonNull final HandleContext handleContext,
+            long amount) {
         final var account = getIfUsable(accountId, accountStore, handleContext.expiryValidator(), INVALID_ACCOUNT_ID);
         final var tokenRel = tokenRelStore.get(accountId, tokenId);
         final var config = handleContext.configuration();
 
+        if (amount != 0) {
```

Comment:
Why do we need this check `amount != 0`

Reply:
Changed it to token.tokenType() == FUNGIBLE_COMMON check.